src/tm_data_types/datum/waveforms/waveform.py

```diff
@@ -56,24 +113,155 @@ def operable_exclusive_metadata(self) -> Dict[str, Any]:
+              suggesting the use of extended_metadata for custom fields
+            - This method is used internally by file format classes to convert
+              between different metadata schemas
         """
-        remapped_dict = {
-            lookup[key]: (val if not isinstance(val, Enum) else val.value)
-            for key, val in data.items()
-            if key in lookup or not drop_non_existant
-        }
+        remapped_dict = {}
+        unknown_keys = []
+        
+        for key, value in data.items():
+            if key in lookup:
+                remapped_dict[lookup[key]] = value
+            else:
+                if not drop_non_existant:
+                    remapped_dict[key] = value
+                    unknown_keys.append(key)
+        
+        # Provide helpful warning for unknown keys
+        if unknown_keys and not drop_non_existant:
+            import warnings
+            warnings.warn(
+                f"Unknown metadata fields {unknown_keys} will be preserved but may not be "
+                f"supported by all file formats. Consider using extended_metadata for custom fields.",
+                UserWarning,
+                stacklevel=2
+            )
+        
         return remapped_dict
 
+    extended_metadata: Optional[Dict[str, Any]] = None
+    """Custom metadata fields that don't fit into the standard waveform metadata.
+    
+    This dictionary allows you to store arbitrary key-value pairs for custom
+    metadata fields. You can access these fields as attributes on the meta_info
+    object using Python's __getattr__ mechanism.
+    
+    The extended_metadata system is designed for:
+    - Application-specific data (test conditions, equipment info, etc.)
+    - Fields that don't fit into the standard waveform metadata schema
+    - Temporary or experimental metadata fields
+    
+    Example:
+        >>> meta_info = WaveformMetaInfo()
+        >>> meta_info.extended_metadata = {
+        ...     "pattern_type": "prbs7",
+        ...     "encoding_type": "nrz", 
+        ...     "symbol_rate": 10e9,
+        ...     "test_equipment": "MSO54"
+        ... }
+        >>> print(meta_info.pattern_type)  # "prbs7"
+        >>> print(meta_info.test_equipment)  # "MSO54"
+        
+    Alternative approach using set_custom_metadata():
+        >>> meta_info.set_custom_metadata(
+        ...     pattern_type="prbs7",
+        ...     test_equipment="MSO54"
+        ... )
+        
+    File Format Compatibility:
+        - WFM files: All custom metadata is preserved
+        - CSV files: Custom metadata is dropped (only standard fields included)
+        - MAT files: Custom metadata may be lost depending on implementation
+        
+    Note:
+        If extended_metadata is None or empty, accessing custom fields will
+        raise an AttributeError with helpful guidance on how to add the field.
+    """
+
+    def __getattr__(self, name: str) -> Any:
```

Comment:
Is this for the entire dataclass or just the extended metadata? Won't this break normal getattr calls on the dataclass? What about the attributes that aren't part of the extended metadata?

Reply:
Thinking specifically about subclasses.